Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp

```diff
@@ -143,6 +143,7 @@ std::unique_ptr<AcceleratedBackingStoreDMABuf> AcceleratedBackingStoreDMABuf::cr
 
 AcceleratedBackingStoreDMABuf::AcceleratedBackingStoreDMABuf(WebPageProxy& webPage)
     : AcceleratedBackingStore(webPage)
+    , m_fenceMonitor([this] { gtk_widget_queue_draw(m_webPage.viewWidget()); })
```

Comment:
You need to wait with using the new dmabuf until the fence signals. Other parts of GTK may have queued a redraw in the meantime (going to the overview or resizing the window for example).

You also need to ensure nobody writes to the dmabuf until GTK has finished using it, which means not signaling the dmabuf until the GdkDmabufTexture's destroy notify runs.

Reply:
> You need to wait with using the new dmabuf until the fence signals. Other parts of GTK may have queued a redraw in the meantime (going to the overview or resizing the window for example).

That's what the patch does.

> You also need to ensure nobody writes to the dmabuf until GTK has finished using it, which means not signaling the dmabuf until the GdkDmabufTexture's destroy notify runs.

We don't exactly do this, we release the buffer right before rendering the next one, I'll check to switch to do it in the destroy notify callback.